LibraryManagement.cpp

```diff
@@ -2,342 +2,332 @@
 #include <string>
 #include <conio.h>
 #include <cstdlib>
+#include <fstream>
 #include <ctime>
+#include <cctype>
 #include "book.h"
 #include "user.h"
 #include "bookmark.h"
 #include "Menu.h"
-
-
 using namespace std;
-
-bool isLoggedIn() {
-    static bool loggedIn = false;
-    return loggedIn;
-}
-
 Menu menu;
-Bookmark bookMark;
-
 int main() {
-	bool quit{false};
-int choose;
-	string username,password;
-    bool isPremium=false;
-    
-
-    cout<<"Enter username: ";
-    cin>>username;
-    while(password.length() < 8) {
-            cout<<"Enter password: ";
-            cin>>password;
-            if(password.length() < 8) {
-                cout<< "Password must contain 8 or more characters..\n";
-            }
-        }
-
-    char userType;
-    cout<<"Select user type(P for premium, R for regular): ";
-    cin>>userType;
-
-
-    if(userType=='P' || userType=='p') {
-    	isPremium=true;
-    	cout<<"Welcome, "<<username<<"!"<<endl;
-    	cout<<"You are a premium user."<<endl;
-    	menu.displayMainMenu();
-        cin>>choose;
-
+    bool shouldLeaveApp = false;
+    Bookmark *bookmark = new Bookmark();
```

Comment:
Hmm why do you need a pointer here?

Reply:
In C++, we use the "new" keyword to dynamically allocate memory for an object and create a pointer to it. Therefore, when we create a new object with a constructor, we must use a pointer to store its address